src/mesh/mesh_output.C

```diff
@@ -52,17 +52,7 @@ void MeshOutput<MT>::write_equation_systems (const std::string & fname,
       // been properly renumbered...
       libmesh_assert(!my_mesh.allow_renumbering());
 
-      libmesh_do_once(libMesh::out <<
-                      "Warning:  This MeshOutput subclass only supports meshes which are contiguously renumbered!"
-                      << std::endl;);
-
-      my_mesh.allow_renumbering(true);
-
-      my_mesh.renumber_nodes_and_elements();
-
-      // Not sure what good going back to false will do here, the
-      // renumbering horses have already left the barn...
-      my_mesh.allow_renumbering(false);
+      libmesh_error_msg("This MeshOutput subclass only supports meshes which are contiguously numbered!");
```

Comment:
Hmm... I think in our internal code, we may be relying on being allowed to get away with renumbering even when a mesh is set to `allow_renumbering(false)`. At least, I see the "This MeshOutput class only supports..." warnings in our logfiles. I think it happens right before we do exodus output, so I'm worried that if you remove this "hack" we'll attempt to write exodus files with non-contiguous numbering (from stitched meshes) and then that will fail? I'd like to try out this change and see if it breaks any of our tests...

Reply:
OK, while I was composing that beautiful paragraph of prose, it looks like you closed the PR :smile:

Reply:
I'd say do try it out, because the whole reason for this PR is that I ran across a bug where a problem was setting `allow_renumbering(false)` and *actually really meant it* ... but as per my other comment, I think the real solution here is going to be to finally just support ExodusII output of meshes with sparse numbering.